src/paint/paint.c

```diff
@@ -959,10 +977,64 @@ void paint_quadrant_ps() {
 			}
 		}
 
-		if (ps->flags & PAINT_STRUCT_FLAG_IS_MASKED)
-			gfx_draw_sprite_raw_masked(dpi, x, y, image_id, ps->colour_image_id);
-		else
-			gfx_draw_sprite(dpi, image_id, x, y, ps->tertiary_colour);
+		if (pPaintBoundingBoxes && dpi->zoom_level == 0) {
+			uint8 colour = BoundBoxDebugColours[ps->sprite_type];
```

Comment:
The problem is that it would need to be two calls, because it draws the
foremost line afterwards.

Op ma 25 jul. 2016 om 13:41 schreef Duncan <notifications@github.com>

> In src/paint/paint.c
> <https://github.com/OpenRCT2/OpenRCT2/pull/3838#discussion_r72049370>:
>
> > @@ -959,10 +977,64 @@ void paint_quadrant_ps() {
> >  			}
> >  		}
> >
> > -		if (ps->flags & PAINT_STRUCT_FLAG_IS_MASKED)
> > -			gfx_draw_sprite_raw_masked(dpi, x, y, image_id, ps->colour_image_id);
> > -		else
> > -			gfx_draw_sprite(dpi, image_id, x, y, ps->tertiary_colour);
> > +		if (pPaintBoundingBoxes && dpi->zoom_level == 0) {
> > +			uint8 colour = BoundBoxDebugColours[ps->sprite_type];
>
> Make this into a separate function call.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenRCT2/OpenRCT2/pull/3838/files/784961d366b0a5edc315ce24d3e5f30da14a573e#r72049370>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAtBQ6eZFnbhY9TiCwq4n4BksazWsOGVks5qZKDlgaJpZM4Iw-Tm>
> .
>